Robot.cpp

```diff
@@ -0,0 +1,101 @@
+#include "Robot.h"
+
+Robot::Robot(double robotLength, double robotWidth, double wheelRadius, Camera & camera, double robotPosTol, double armPosTol) 
+    : robotLength(robotLength), robotWidth(robotWidth), wheelRadius(wheelRadius), camera(camera), robotPosTol(robotPosTol), armPosTol(armPosTol) {
+    // pid terms need to be determined experimentally
+    drive[frontLeft] = MotorController(10, 1, 1, 5, 0);
+    drive[backLeft] = MotorController(10, 1, 1, 5, 0);
+    drive[frontRight] = MotorController(10, 1, 1, 5, 0);
+    drive[backRight] = MotorController(10, 1, 1, 5, 0);
+
+    servoArm[x] = MotorController(10, 1, 1, 5, 0);
+    servoArm[y] = MotorController(10, 1, 1, 5, 0);
+    servoArm[z] = MotorController(10, 1, 1, 5, 0);
+
+    robotPosition = Point(0, 0, 0);
+    armPosition = Point(0, 0, 0);
+}
+
+Point Robot::getRobotPosition(){
+    return robotPosition;
+}
+
+Point Robot::getArmPosition(){
+    return armPosition;
+}
+
+// how are we going to do this? 
+// should it just be called at the end of each robot move function? 
+void Robot::updateRobotPosition(){}
+void Robot::updateArmPosition(){}
+
+// From MTE 544 Control Lecture Slide 9
+double calculate_radius(double delta_x, double delta_y){
+    double L = sqrt(delta_x*delta_x+delta_y*delta_y);
+
+    return L*L/2*std::abs(delta_y);
+}
+
+// From MTE 544 Modeling III IV Lecture Slide 16
+// Pass in -w for left wheel
```

Comment:
this assumes our motor signals have a CCW conversion, right? Should note that somewhere

Reply:
just added a comment for this here, unless you think the comment should be somewhere else